contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/json/JsonTableGroupScan.java

```diff
@@ -106,7 +115,11 @@ private void init() {
       Configuration conf = new Configuration();
       Table t = MapRDB.getTable(scanSpec.getTableName());
       TabletInfo[] tabletInfos = t.getTabletInfos(scanSpec.getCondition());
-      tableStats = new MapRDBTableStats(conf, scanSpec.getTableName());
+
+      // Fetch tableStats only once and cache it.
+      if (tableStats == null) {
```

Comment:
This can probably be removed if we call clone(). However, it may be a useful check if we end up calling it from some other code-paths. Maybe add some logging to ensure we are not recreating the tableStats?